src/main/java/org/sagebionetworks/web/client/widget/entity/browse/EntityFinderV2Impl.java

```diff
@@ -0,0 +1,348 @@
+        this.view = view;
+        this.globalApplicationState = globalApplicationState;
+        this.authenticationController = authenticationController;
+        this.synAlert = synAlert;
+        this.jsClient = jsClient;
+        this.selectedEntities = new ArrayList<>();
+        view.setPresenter(this);
+        view.setSynAlert(synAlert.asWidget());
+    }
+
+    private EntityFinderV2Impl(Builder builder) {
+        this.selectedEntities = new ArrayList<>();
+
+        // Dependencies injected into the builder
+        this.view = builder.view;
+        this.globalApplicationState = builder.globalApplicationState;
+        this.synAlert = builder.synAlert;
+        this.jsClient = builder.jsClient;
+
+        this.view.setPresenter(this);
+        this.view.setSynAlert(this.synAlert.asWidget());
+
+        // Configuration
+        if (builder.modalTitle != null) {
+            this.view.setModalTitle(builder.modalTitle);
+        }
+        if (builder.promptCopy != null) {
+            this.view.setPromptCopy(builder.promptCopy);
+        }
+        if (builder.confirmButtonCopy != null) {
+            this.view.setConfirmButtonCopy(builder.confirmButtonCopy);
+        }
+
+        modalTitle = builder.modalTitle;
+        promptCopy = builder.promptCopy;
+        selectedCopy = builder.selectedCopy;
+        modalTitle = builder.modalTitle;
+        multiSelect = builder.multiSelect;
+        initialContainerId = builder.initialContainerId;
+        selectedHandler = builder.selectedHandler;
+        selectedMultiHandler = builder.selectedMultiHandler;
+        selectableFilter = builder.selectableFilter;
+        visibleFilter = builder.visibleFilter;
+        showVersions = builder.showVersions;
+        renderComponent();
+    }
+
+    public static class Builder implements EntityFinder.Builder {
+        private EntityFinderV2View view;
+        GlobalApplicationState globalApplicationState;
+        AuthenticationController authenticationController;
+        private SynapseAlert synAlert;
+        private SynapseJavascriptClient jsClient;
+
+        private SelectedHandler<Reference> selectedHandler = (selected, finder) -> {
+        };
+        private SelectedHandler<List<Reference>> selectedMultiHandler = (selected, finder) -> {
+        };
+
+        private boolean showVersions = false;
+        private EntityFilter visibleFilter = ALL;
+        private EntityFilter selectableFilter = ALL;
+
+        private boolean multiSelect = false;
+        private String initialContainerId = null;
+
+        private String modalTitle = "Find in Synapse";
+        private String promptCopy = "";
+        private String selectedCopy = "Selected";
+        private String confirmButtonCopy = "Select";
+
+        @Inject
+        public Builder(EntityFinderV2View view, GlobalApplicationState globalApplicationState, AuthenticationController authenticationController, SynapseAlert synAlert, SynapseJavascriptClient jsClient) {
```

Comment:
An alternative approach, that would avoid the dependency injection mess, would be to create a builder for an EntityFinderConfiguration object instead (then I think you would have one configure(EntityFinderConfiguration) method to rule them all).  Then you could pull the builder into it's own class, and not require injection of the builder (just the EntityFinderV2).